api/src/test/java/com/epam/pipeline/manager/HierarchicalEntityManagerTest.java

```diff
@@ -16,348 +16,312 @@
+        mockUserContext();
+        initAclEntity(dockerRegistryList);
+        initAclEntity(registry);
+        initAclEntity(toolGroup);
+        initAclEntity(tool, AclPermission.READ);
+
+        final Map<AclClass, List<AbstractSecuredEntity>> available = hierarchicalEntityManager
+                .loadAvailable(new AclSid(SIMPLE_USER, true), null);
+
+        assertThat(available.size()).isEqualTo(1);
+        assertThat(available.get(AclClass.TOOL).get(0).getMask()).isEqualTo(READ_PERMISSION);
     }
 
     @Test
-    @WithMockUser(username = USER)
-    public void testLoadingByRoleSidWorks() {
-        DockerRegistry registry = registryManager.create(
-                ObjectCreatorUtils.createDockerRegistryVO(TEST_NAME, USER, USER));
-        ToolGroup toolGroup = toolGroupManager.create(
-                ObjectCreatorUtils.createToolGroup(TEST_NAME, registry.getId()));
-        Tool tool = toolManager.create(
-                ObjectCreatorUtils.createTool(TEST_NAME, toolGroup.getId()), false);
-        grantPermission(tool.getId(), AclClass.TOOL, DefaultRoles.ROLE_USER.getName(),
-                false, AclPermission.READ.getMask());
-
-        Folder folder = createFolder(TEST_NAME, null);
-        RunConfiguration runConfiguration = createRunConfiguration(TEST_NAME, folder.getId());
-        grantPermission(runConfiguration.getId(), AclClass.CONFIGURATION, DefaultRoles.ROLE_USER.getName(),
-                false, AclPermission.READ.getMask());
-
-        Map<AclClass, List<AbstractSecuredEntity>> available = hierarchicalEntityManager.loadAvailable(
-                new AclSid(DefaultRoles.ROLE_USER.getName(), false), null);
-        Assert.assertEquals(2, available.size());
-        Assert.assertEquals((int) available.get(AclClass.TOOL).get(0).getMask(), AclPermission.READ.getMask());
+    @WithMockUser
+    public void shouldLoadByRoleSid() {
+        final DockerRegistry registry = getDockerRegistry(ID, ANOTHER_SIMPLE_USER);
+        final Tool tool = getTool(ID, ANOTHER_SIMPLE_USER);
+        final ToolGroup toolGroup = getToolGroup(ID, ANOTHER_SIMPLE_USER);
+        final DockerRegistryList dockerRegistryList = getDockerRegistryList(ID, ANOTHER_SIMPLE_USER, registry);
+        final Folder folder = getFolder(ANOTHER_SIMPLE_USER);
+        final RunConfiguration runConfiguration = getRunConfiguration(ID, ANOTHER_SIMPLE_USER);
+        toolGroup.setTools(Collections.singletonList(tool));
+        registry.setGroups(Collections.singletonList(toolGroup));
+        folder.setConfigurations(Collections.singletonList(runConfiguration));
+        doReturn(dockerRegistryList).when(mockRegistryManager).loadAllRegistriesContent();
+        doReturn(folder).when(mockFolderManager).loadTree();
+        mockUserContext();
+        initAclEntity(dockerRegistryList);
+        initAclEntity(folder);
+        initAclEntity(registry);
+        initAclEntity(toolGroup);
+        initAclEntity(tool,
```

Comment:
Let's grant permissions for role `ROLE_USER` (as it done in `shouldLoadByRoleSidWhenLoadForUser` test)